ros/src/tl_detector/tl_detector.py

```diff
@@ -268,18 +268,76 @@ def get_light_state(self, light):
         # Get classification
         return self.light_classifier.get_classification(img_traffic_light)
 
-    def get_tl_waypoints(self):
-        """ Converts array self.lights with trafic
-        light positions to get_closest_waypoint_idx array with PoseStamped
-        traffic light waypoints
-
+    def get_tl_waypoints_idx(self):
+        """ Converts array self.lights with trafic light positions to
+            tl_waypoints_idx array with traffic light waypoint indexes
         """
 
         for light in self.lights:
-            self.tl_waypoints.append(
+            self.tl_waypoints_idx.append(
                 self.get_closest_waypoint_idx(light.pose.pose)
             )
 
+    def get_distance_in_track(self, a, b, track_length):
+        """Finds the sortest signed distance between a and b,
+            (b - a), considering the track_length
+
+            Example 1
+            a = car = 190
+            b = light = 10
+            output = 20 (in front of the car)
+
+            Example 2
+            a = car = 10
+            b = light = 190
+            output = -20 (behind the car)
+        """
+
+        output = b - a
+
+        if (output < -0.5 * track_length):
+            output += track_length
+        elif (output > 0.5 * track_length):
+            output -= track_length
+
+        return output
+
+    def get_closest_tl_wp_idx(self, car_pos_wp_idx, searching_dist_tl):
+        """ Finds closest traffic light waypoint in searching range
+
+        Args:
+            car_pos_wp_idx (Integer): current position of the car
+            searching_dist_tl: number of waypoints that will be searched
+            for a traffic light
+
+        Returns:
+            light_wp_idx: waypoint index of closest traffic light in range
+
+        """
+
+        light_wp_idx = -1
+        # In case that array with traffic light waypoints does not exist,
+        # create it
+        if(self.tl_waypoints_idx == []):
+            self.get_tl_waypoints_idx()
+        last_waypoint_idx = len(self.waypoints.waypoints) - 1
```

Comment:
When passed to the function `get_distance_in_track`, you should pass the actual length (as number of waypoints), not the index to the last element. So it should not have the `-1` there. Maybe you can call it `track_length` or similar as well?

Also, `car_distance_from_last_wp` is not used anymore so can be removed